packages/lib/models/utils/readOnly.ts

```diff
@@ -75,13 +76,17 @@ export const checkIfItemCanBeAddedToFolder = async (itemType: ModelType, Folder:
 // extra `sharePermissionCheckOnly` boolean to do the check for one case or the other. A bit of a
 // hack but good enough for now.
 export const itemIsReadOnlySync = (itemType: ModelType, changeSource: number, item: ItemSlice, userId: string, shareState: ShareState, sharePermissionCheckOnly = false): boolean => {
-	checkObjectHasProperties(item, sharePermissionCheckOnly ? ['share_id'] : ['share_id', 'deleted_time']);
+	if (!sharePermissionCheckOnly && isTrashableItem({ id: item.id, type_: itemType })) {
+		checkObjectHasProperties(item, ['deleted_time']);
+	}
 
 	// Item is in trash
 	if (!sharePermissionCheckOnly && item.deleted_time) return true;
 
 	if (!needsShareReadOnlyChecks(itemType, changeSource, shareState)) return false;
 
+	checkObjectHasProperties(item, ['share_id']);
```

Comment:
This check was moved from the top of the function to after `needsShareReadOnlyChecks`. This matches the behavior of the function prior to [this commit](https://github.com/laurent22/joplin/commit/f19b1c536439bc0a9f04253203371c15b60f65b3#diff-aecd8fde4b82749dc0274923415a72d3b1bab247aacf2d7bff82036be9c681e0L70).

This seems to be necessary to allow the new resource-related test to pass (see `readOnly.test.ts`).